docs/stacks-guide-rook.md

```diff
@@ -0,0 +1,830 @@
+rook-crds             CustomResourceDefinition   ybclusters.yugabytedb.rook.io   yugastore-k8s   Submitted
+rook-namespace        Namespace                  rook-yugabytedb-system          yugastore-k8s   Submitted
+rook-serviceaccount   Deployment                 rook-yugabytedb-operator        yugastore-k8s   Submitted
+```
+
+It is not unusual for the `rook-yugabytedb-operator` to fail on first attempt.
+It is reliant on the `ClusterRole` and will fail if it is not present when the
+container starts. However, the Crossplane `workload` controllers will take care
+of recreating it for us!
+
+## Deploy Yugastore alongside YugabyteDB
+
+Now that we have a GKE cluster up and running, we can create our YugabyteDB
+cluster and install Yugastore alongside it.
+
+```bash
+cat > yugastore.yaml <<EOF
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  name: yugastore-db
+  namespace: yugastore-app
+  labels:
+    app: yugastore
+spec:
+  classSelector:
+    matchLabels:
+      app: yugastore
+  writeConnectionSecretToRef:
+    name: yugastore-db-secret
+---
+apiVersion: workload.crossplane.io/v1alpha1
+kind: KubernetesApplication
+metadata:
+  name: yugastore
+  namespace: yugastore-app
+  labels:
+    app: yugastore
+spec:
+  resourceSelector:
+    matchLabels:
+      app: yugastore
+  clusterSelector:
+    matchLabels:
+      app: yugastore
+  resourceTemplates:
+  - metadata:
+      name: yugastore-namespace
+      labels:
+        app: yugastore
+    spec:
+      template:
+        apiVersion: v1
+        kind: Namespace
+        metadata:
+          name: rook-yugastore
+          labels:
+            app: yugastore
+  - metadata:
+      name: yugastore-deployment
+      labels:
+        app: yugastore
+    spec:
+      template:
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+          namespace: rook-yugastore
+          name: yugastore
+          labels:
+            app: yugastore
+        spec:
+          selector:
+            matchLabels:
+              app: yugastore
+          template:
+            metadata:
+              labels:
+                app: yugastore
+            spec:
+              containers:
+                - name: yugastore
+                  image: gcr.io/crossplane-playground/yugastore:latest
+                  imagePullPolicy: Always
+                  command: ["/usr/local/yugastore/bin/start-for-crossplane.sh"]
+                  env:
+                  - name: DB_HOST
+                    value: "yb-tserver-hello-ybdb-cluster-1.yb-tservers-hello-ybdb-cluster.rook-yugabytedb.svc.cluster.local"
+                  - name: APP_HOST
+                    value: "localhost"
+                  - name: APP_PORT
+                    value: "3001"
+                  ports:
+                    - containerPort: 3001
+                      name: yugastore
+                  livenessProbe:
```

Comment:
This health check causes the application to be restarted if it was unable to seed the database. This works fine but feels sort of hacky (I literally create the `healthy` file following the db init in the bootscript: https://github.com/hasheddan/yugastore/blob/5d4d887c93cc33fdabc9ef9365fb6c39f715d437/bin/start-for-crossplane.sh#L20). If anyone has a better idea, I would welcome it. I got lost wrangling with the javascript this morning: https://github.com/hasheddan/yugastore/blob/waitfordb/models/yugabyte/db_init.js

cc @jbw976 @negz @rbwsam

Reply:
Follow up on this: apparently the app has a few holes, so we would be working with them to iron them out. I think this is fine as the current configuration still brings up a fun little app to demo, but certainly wouldn't call it production ready or anything.